File: alligator/src/main/java/me/aartikov/alligator/converters/DialogFragmentConverter.java
package me.aartikov.alligator.converters;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import me.aartikov.alligator.Screen;


/**
 * Converts a screen to a dialog fragment and vice versa.
 *
 * @param <ScreenT> screen type
 */
public interface DialogFragmentConverter<ScreenT extends Screen> {

	@NonNull
	DialogFragment createDialogFragment(@NonNull ScreenT screen);

	@NonNull
	ScreenT getScreen(@NonNull DialogFragment fragment);
}

File: alligator/src/main/java/me/aartikov/alligator/converters/DialogFragmentConverter.kt
package me.aartikov.alligator.converters

import androidx.fragment.app.DialogFragment
import me.aartikov.alligator.Screen

/**
 * Converts a screen to a dialog fragment and vice versa.
 *
 * @param <ScreenT> screen type
</ScreenT> */
interface DialogFragmentConverter<ScreenT : Screen?> {
    fun createDialogFragment(screen: ScreenT): DialogFragment
    fun getScreen(fragment: DialogFragment): ScreenT
}